src/main/java/io/tackle/pathfinder/controllers/TranslatorResource.java

```diff
@@ -0,0 +1,28 @@
+package io.tackle.pathfinder.controllers;
+
+import io.tackle.pathfinder.dto.translator.TranslateDBDto;
+import io.tackle.pathfinder.services.TranslatorSvc;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/translator")
+public class TranslatorResource {
+    @Inject
+    TranslatorSvc translatorSvc;
+
+    @POST
+    @Path("/translate")
+    public Response translateText(@NotNull TranslateDBDto translateDBDto) {
+        String key = translatorSvc.getKey(translateDBDto.getTable(), translateDBDto.getId(), translateDBDto.getField());
+        translatorSvc.addOrUpdateTranslation(key, translateDBDto.getText(), translateDBDto.getLanguage());
+        return Response
```

Comment:
Is there a scenario where adding the translation may have failed but no exception has been thrown? In that case we would need a different Response to be built.

Reply:
I can not think of a scenario where these lines can throw an exception other than some Database connection exception. In case any DB associated exception is thrown then this ExceptionMapper should handle it: https://github.com/konveyor/tackle-pathfinder/blob/main/src/main/java/io/tackle/pathfinder/exception/GlobalExceptionMapper.java